### ISSUE TITLE
react/display-name false positives for named exports using React.memo

### ISSUE THREAD

**Issue by mskelton** [yannickcr/eslint-plugin-react#2445]
The `react/display-name` rule is giving false positives when using the following code.

```js
export const MyComponent = React.memo(function MyComponent() {
})
```

The same error occurs if you have different names for the component and export.

```js
export const ExportedComponent = React.memo(function MyComponent() {
})
```

**Comment by ljharb** [comment]
Which errors do you get?

**Comment by pawelnvk** [comment]
@markypython could you provide more details? I'm not sure what's the issue here. I investigated briefly these snippets. I observed that following snippet raises an error no matter what `ignoreTranspilerName`'s value was and I feel that shouldn't be an issue, since in devtools only relevant names are these coming out of passed component. 
```jsx
export const MyComponent = React.memo(function MyComponent() {});
```
Nevertheless I would say that following snippet should be considered as misleading, since the name of component in devtools will be `MyComponent`. Yet I'm not sure if this rule should be responsible for this. What's your opinion on that @ljharb?
```jsx
export const ExportedComponent = React.memo(function MyComponent() {});
```

**Comment by ljharb** [comment]
Yes, the rule should accept this since memo will pull its displayName from the component.

**Comment by markypython** [comment]
Sorry for the delay in responding.  The error is `Component definition is missing display nameeslint(react/display-name)`.

**Comment by pawelnvk** [comment]
Ok. I'll try to work on this one.

**Comment by giovannigiordano** [comment]
@pawelnvk any update on this?
